ChartsRealm/Classes/Data/RealmBaseDataSet.swift

```diff
@@ -21,272 +21,284 @@ open class RealmBaseDataSet: ChartBaseDataSet
     {
         fatalError("RealmBaseDataSet is an abstract class, you must inherit from it. Also please do not call super.initialize().")
     }
-    
+
     public required init()
     {
         super.init()
-        
+
         // default color
         colors.append(NSUIColor(red: 140.0/255.0, green: 234.0/255.0, blue: 255.0/255.0, alpha: 1.0))
-        
+
         initialize()
     }
-    
+
     public override init(label: String?)
     {
         super.init()
-        
+
         // default color
         colors.append(NSUIColor(red: 140.0/255.0, green: 234.0/255.0, blue: 255.0/255.0, alpha: 1.0))
-        
+
         self.label = label
-        
+
         initialize()
     }
-    
+
+    // MARK: Realm for Objective-C
+
     @objc public init(results: RLMResults<RLMObject>?, xValueField: String?, yValueField: String, label: String?)
     {
         super.init()
-        
+
         // default color
         colors.append(NSUIColor(red: 140.0/255.0, green: 234.0/255.0, blue: 255.0/255.0, alpha: 1.0))
-        
+
         self.label = label
-        
-        _results = results
+
+        _results = results as? RLMResults<RLMObjectBase>
```

Comment:
@jjatie Not sure if you can see this. Need your help here

I learned that `Results<SubObject>` cannot be converted to `Results<Object>`, per https://stackoverflow.com/questions/48125923/realmswift-cannot-cast-resultssomeojbect-to-resultsobject/48126909?noredirect=1#comment83231956_48126909

However, here I can cast `RLMResults<RLMObject>` to `results as? RLMResults<RLMObjectBase>`, without problem.

I am kind of confused here. Any idea? Thanks.

Reply:
I'll take a look this week.

Reply:
further experiments show that, I could even pass a `Result<SomeObejct>` into `converted = ObjectiveCSupport.convert(object: results!) as? RLMResults<RLMObject>`, and it can even convert `Result<SomeObejct>` to `RLMResults<RLMObject>`, even `SomeObejct` and `RLMObject` has no relationship. 
 I asked this in https://github.com/realm/realm-cocoa/issues/5555, to see if they add any.

Reply:
it's said in realm/realm-cocoa#5555 Object is using Swift generics, while RLMObject is Objc generics, so they are different